project-base/storefront/components/Pages/Order/TransportAndPayment/TransportAndPaymentContent.tsx

```diff
@@ -64,7 +64,7 @@ export const TransportAndPaymentContent: FC = () => {
     };
 
     useEffect(() => {
-        if (!cart?.items.length) {
+        if (cart !== undefined && !cart?.items.length) {
             router.replace(cartUrl);
         }
     }, [cart?.items]);
```

Comment:
Since you check for undefined, you do not need to optionally chain it:

```ts
if (cart !== undefined && !cart.items.length) {
    router.replace(cartUrl);	        
}
```

and since you have to check for undefined, you could even explicitly check for the length, hmm?

Reply:
I need to specificaly check for undefined because I use `null` and `undefined` for purpose to distinguish if cart has been fetched or exist. In case it exist but it's empty, it's null. Otherwise it's undefined. This allows me to show empty cart message or loading state. Length was already discussed.